sources/packages/backend/apps/api/src/services/student-scholastic-standings/student-scholastic-standings.service.ts

```diff
@@ -291,9 +294,11 @@ export class StudentScholasticStandingsService extends RecordDataModelService<St
       // Case a restriction was created, send a notification to the student.
       // Left as the last step to ensure that everything else was processed with
       // success and the notification will not be generated otherwise.
-      if (createdRestriction) {
+      if (createdRestrictions.length) {
         await this.studentRestrictionSharedService.createNotifications(
-          [createdRestriction.id],
+          createdRestrictions.map(
+            (createdRestriction) => createdRestriction.id,
+          ),
```

Comment:
Although minor and not a blocker, it would be better for readability if the map was performed before line 298.